lib/kafka_ex/server.ex

```diff
@@ -209,21 +214,21 @@ defmodule KafkaEx.Server do
     Enum.each(state.brokers, fn(broker) -> KafkaEx.NetworkClient.close_socket(broker.socket) end)
   end
 
-  defp update_consumer_metadata(state, consumer_group), do: update_consumer_metadata(state, consumer_group, 3, 0)
+  defp update_consumer_metadata(state), do: update_consumer_metadata(state, 3, 0)
 
-  defp update_consumer_metadata(state, consumer_group, 0, error_code) do
+  defp update_consumer_metadata(state = %State{consumer_group: consumer_group}, 0, error_code) do
     Logger.log(:error, "Fetching consumer_group #{consumer_group} metadata failed with error_code #{inspect error_code}")
     {%Proto.ConsumerMetadata.Response{error_code: error_code}, state}
   end
 
-  defp update_consumer_metadata(state, consumer_group, retry, _error_code) do
-    consumer_group_metadata_request = Proto.ConsumerMetadata.create_request(state.correlation_id, @client_id, consumer_group)
+  defp update_consumer_metadata(state = %State{consumer_group: consumer_group, correlation_id: correlation_id}, retry, _error_code) do
+    consumer_group_metadata_request = Proto.ConsumerMetadata.create_request(correlation_id, @client_id, consumer_group)
     data = Enum.find_value(state.brokers, fn(broker) -> KafkaEx.NetworkClient.send_sync_request(broker, consumer_group_metadata_request, state.sync_timeout) end)
     response = Proto.ConsumerMetadata.parse_response(data)
     case response.error_code do
-      0 -> {response, %{state | consumer_metadata: response, consumer_group: consumer_group, correlation_id: state.correlation_id + 1}}
+      0 -> {response, %{state | consumer_metadata: response, correlation_id: state.correlation_id + 1}}
```

Comment:
It makes more sense to me to have this function just return the mutated state, and everywhere you have `{consumer_metadata, state} = update_consumer_metadata(state)` you could just have `state = update_consumer_metadata(state)` and then use `state.consumer_metadata`.